airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/NormalizationSummaryCheckActivityImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.sync;
+
+import io.airbyte.persistence.job.JobPersistence;
+import io.airbyte.persistence.job.models.AttemptNormalizationStatus;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class NormalizationSummaryCheckActivityImpl implements NormalizationSummaryCheckActivity {
+
+  private final Optional<JobPersistence> jobPersistence;
+
+  public NormalizationSummaryCheckActivityImpl(final Optional<JobPersistence> jobPersistence) {
```

Comment:
What's the use case for the optional JobPersistence?
If it's related to the data plane worker, feels like we may want to have a follow up to be able to use API instead of direct DB access here, otherwise multi-cloud will run into this issue again.

Reply:
yes that's exactly the reason - i agree, didn't realize while I was writing this but changing this to use the API instead later makes sense. It's not hugely crucial in this case since all that will happen is that normalization will run when it doesn't need to

Reply:
In the end, this PR is still a step forward. I think this also highlights the fact that we need to focus on removing direct DB access from the workers.

cc @jdpgrailsdev in case you were keeping track of the many reasons we should be doing this ;)